### PR TITLE
Spotbugs fixes for `LI_LAZY_INIT_STATIC`, `NP_LOAD_OF_KNOWN_NULL_VALUE`, and `UWF_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD`

### PR DESCRIPTION
**Pull request by basil**
Triages a few more SpotBugs violations.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/util/ProcessTree.java
Comment (basil) [inline core/src/main/java/hudson/util/ProcessTree.java:437]: Fixes `LI_LAZY_INIT_STATIC` for `vetoersExist`. In general, the whole initialization process does not appear to be thread-safe, so synchronizing on the whole method seems safer than doing fine-grained synchronization on `vetoersExist`.

File: core/src/main/java/jenkins/security/MasterToSlaveCallable.java
Comment (basil) [inline core/src/main/java/jenkins/security/MasterToSlaveCallable.java:33]: Chases away `NP_LOAD_OF_KNOWN_NULL_VALUE`.

**Review by StefanSpieker — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by oleg-nenashev — COMMENTED** [review]
I wonder what is the performance impact of `public static synchronized ProcessTree get()`

**Comment by basil** [comment]
> I wonder what is the performance impact of `public static synchronized ProcessTree get()`

Not sure offhand. Is there a specific action item here? In general, I have a hard time when people leave me with reviews without specific next steps.

**Comment by oleg-nenashev** [comment]
@basil asking just in case you have investigated that. I am not blocking the pull request, the 24hrs response time applies to me as to any other contributor. May take a look tomorrow, not blocking otherwise

**Pull request merged**
